processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -258,7 +259,8 @@ public ColumnCapabilities getColumnCapabilities(String columnName)
   protected IncrementalIndex(
       final IncrementalIndexSchema incrementalIndexSchema,
       final boolean deserializeComplexMetrics,
-      final boolean concurrentEventAdd
+      final boolean concurrentEventAdd,
+      final boolean preserveExistingMetrics
```

Comment:
i feel like this definitely needs javadocs/comments to ensure that nothing ever sets this except DruidInputSource, since that is the only case where we should have existing metrics

Reply:
Done